### DISCUSSION TITLE
Everyday Usage: Unable to install WireGuard using Homebrew on macOS High Sierra 10.13.6

### DISCUSSION THREAD

**Discussion by Michsarri** [Homebrew/discussions#5570] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3426911432b93dc46d20d910c2f4fd33cdac62a2
Last commit: 5 days ago
Core tap JSON: 24 Aug 14:54 UTC
Core cask tap JSON: 24 Aug 14:54 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: quad-core 64-bit sandybridge
Clang: 10.0.0 build 1000
Git: 2.17.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.13.6-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You are using macOS 10.13.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Hello everyone,

I'm encountering an issue while trying to install WireGuard on my MacBook, which is running an older version of macOS—High Sierra 10.13.6 (17G14042).

I have successfully installed Homebrew on my system, but when I attempt to install WireGuard using the following command:

brew install wireguard-tools

the installation fails, and I receive the following error message:

Error: wireguard-tools: SHA256 mismatch
Expected: ed3694e808d96720e79e17ae396f89f7c2024da07f3449ff0af8fbc6dbfa7f6a
Actual: 50029ca43196cc3d925c0f0b98a9fcfd7f6c28465122da90ebe5262398f3b31c
File: /Users/alina_voropaeva/Library/Caches/Homebrew/downloads/f1c31ba406a7d0c14cac90f9a2b7da36e41378900fc9b2ba1f48ee9f29b5ad0e--wireguard-go-0.0.20230223.tar.xz
To retry an incomplete download, remove the file above.

I would greatly appreciate any guidance on how to resolve this issue. Has anyone else encountered a similar problem, and if so, how did you fix it?

Thank you in advance for your help!

**Comment by gromgit** [comment]
> Error: `wireguard-tools`: SHA256 mismatch
> [...]
> File: /Users/alina_voropaeva/Library/Caches/Homebrew/downloads/f1c31ba406a7d0c14cac90f9a2b7da36e41378900fc9b2ba1f48ee9f29b5ad0e--`wireguard-go`-0.0.20230223.tar.xz

One of these things is not like the other...

Anyway, I confirmed that the current `wireguard-go` source tarball does indeed have a different checksum than before, and have initiated a query upstream to find out more.

**Reply by Michsarri** [reply]
Thank you for looking into this and confirming the checksum issue. I really appreciate your help in reaching out to the upstream maintainers to get more information. I'll wait for your update. Please let me know if there’s anything else I can do on my end.

Thanks again for your support!

**Comment by gromgit** [comment]
Upstream confirmed the tarball checksums changed because they upgraded `xz` on the release system. They've since reverted to the previous version of `xz`, and the current Homebrew checksums match again.
